### PR TITLE
Update changelog in prep for 5.0

### PR DESCRIPTION
**Pull request by aaronayres35**
This PR updates the changelog in preparation for the 5.0 release.

Changes made after the upcoming rc and the real release will be added in a later PR

### CODE REVIEW

**Comment by aaronayres35** [comment]
There is a very large number of changes for this release.  I think I am going to add a new "removals" section given there are so many

File: CHANGES.txt [hunk 1]
```diff
@@ -2,6 +2,151 @@
...
 ===============
 
+Release 5.0.0
+-------------
+
+New features/Improvements
+
+* Add rectangular selection tool (#482)
+* Add missing property index_range to 1D plots (#491)
+* Feature: support renderer traits (#514)
```
Comment (rahulporuri) [inline CHANGES.txt:12]:
```suggestion
* Support renderer traits (#514)
```

File: CHANGES.txt [hunk 2]
```diff
@@ -2,6 +2,151 @@
...
+
+Fixes
+
+* Fix typo in the error name (#484)
+* Fix attempts to get the .shape attribute of a list or tuple. (#487)
```
Comment (rahulporuri) [inline CHANGES.txt:17]:
```suggestion
* Fix attempts to get the shape attribute of a list or tuple. (#487)
```

File: CHANGES.txt [hunk 3]
```diff
@@ -2,6 +2,151 @@
...
+* Fix scrollbar demo and update docstring (#489)
+* Fix stacked bar plot example, by using the correct starting_value (#498)
+* Limit the min zoom out factor along x axis to prevent date range error (#500)
+* Do not expose 'auto' for infinite series (#501)
+* Fix zoomout issues (#505)
+* Fix: Zoom not working with contour plots (#506)
```
Comment (rahulporuri) [inline CHANGES.txt:23]:
```suggestion
* Fix zoom not working with contour plots (#506)
```

File: CHANGES.txt [hunk 4]
```diff
@@ -2,6 +2,151 @@
...
+* Fix and clean up multiaxis.py example (#513)
+* FIX: Enforce underscore for click command (#523)
```
Comment (rahulporuri) [inline CHANGES.txt:25]:
```suggestion
* Enforce underscore for click command (#523)
```

File: CHANGES.txt [hunk 5]
```diff
@@ -2,6 +2,151 @@
...
+* Handle index=None in ScatterInspector (#540)
+* have ArrayDataSource.get_data return empty(shape=0,)), not 0.0 if _data is None (#578)
```
Comment (rahulporuri) [inline CHANGES.txt:27]:
```suggestion
* Have ArrayDataSource.get_data return empty(shape=0,)), not 0.0 if _data is None (#578)
```

File: CHANGES.txt [hunk 6]
```diff
@@ -2,6 +2,151 @@
...
+* return correct sized tuples of all 0s not None if image plot resized too small (#589)
```
Comment (rahulporuri) [inline CHANGES.txt:28]:
```suggestion
* Return correct sized tuples of all 0s not None if image plot resized too small (#589)
```

File: CHANGES.txt [hunk 7]
```diff
@@ -2,6 +2,151 @@
...
+* Resolve the NotifierNotFound error in testsuite (#601)
+* Catch nans before they cause crash in auto_ticks (#636)
+* fix windows/pyqt test_dont_crash_on_click failure (#663)
```
Comment (rahulporuri) [inline CHANGES.txt:31]:
```suggestion
* Fix windows/pyqt test_dont_crash_on_click failure (#663)
```

File: CHANGES.txt [hunk 8]
```diff
@@ -2,6 +2,151 @@
...
+* Use draw_marker_at_points if possible otherwise use draw_path_at_points in ColormappedScatterPlot (#672)
+* Fix a couple of demos (#676)
+* use plot_range_event (aka event.new) not just event in stock_prices (#728)
```
Comment (rahulporuri) [inline CHANGES.txt:34]:
```suggestion
* Use plot_range_event (aka event.new) not just event in stock_prices (#728)
```

File: CHANGES.txt [hunk 9]
```diff
@@ -2,6 +2,151 @@
...
+* fix broken test and handle case when _data is None in ArrayDataSource.get_data_mask() (#731)
```
Comment (rahulporuri) [inline CHANGES.txt:35]:
```suggestion
* Fix broken test and handle case when _data is None in ArrayDataSource.get_data_mask() (#731)
```

File: CHANGES.txt [hunk 10]
```diff
@@ -2,6 +2,151 @@
...
+* replace [] default for data with None (#734)
```
Comment (rahulporuri) [inline CHANGES.txt:36]:
```suggestion
* Replace [] default for data with None (#734)
```

File: CHANGES.txt [hunk 11]
```diff
@@ -2,6 +2,151 @@
...
+
+Documentation/examples
+
+* Documentation Refactor (#462)
+* ENH: Setup online redirect and proper index page for docset (#517)
```
Comment (rahulporuri) [inline CHANGES.txt:41]:
```suggestion
* Setup online redirect and proper index page for docset (#517)
```

File: CHANGES.txt [hunk 12]
```diff
@@ -2,6 +2,151 @@
...
+* Replace all jet color map use in examples with viridis (#557)
+* Update various comments and docstrings (#567)
+* Clean up updates for basic_elements docs pages (#570)
+* Update docs conf.py - use Enthought sphinx template if available. (#571)
```
Comment (rahulporuri) [inline CHANGES.txt:45]:
```suggestion
* Use Enthought Sphinx Theme if available. (#571)
```

File: CHANGES.txt [hunk 13]
```diff
@@ -2,6 +2,151 @@
...
+* fix broken image links on docs/source/user_manual/quickstart.rst (#588)
```
Comment (rahulporuri) [inline CHANGES.txt:46]:
```suggestion
* Fix broken image links on quickstart page (#588)
```

File: CHANGES.txt [hunk 14]
```diff
@@ -2,6 +2,151 @@
...
+* Update modules_and_classes.rst (#652)
+* Remove annotated examples from the usermanual (#653)
+* Use sphinx todo for currently empty sections of docs (#654)
+* Remove optparse from example (#655)
+* Delete the quickstart page from the docs (#668)
+* Remove references to chaco.shell from the documentation (#669)
+* Update van der waal example (#680)
+* Remove scipy tutorial content (#681)
+* Remove chaco_talk_apr_2008 from docs (#682)
+* Update hyetopgraph example (#687)
```
Comment (rahulporuri) [inline CHANGES.txt:60]:
```suggestion
* Update hyetograph example (#687)
```

File: CHANGES.txt [hunk 15]
```diff
@@ -2,6 +2,151 @@
...
+* Remove unused related images from docs (#712)
```
Comment (rahulporuri) [inline CHANGES.txt:61]:
```suggestion
* Remove unused images from codebase (#712)
```

File: CHANGES.txt [hunk 16]
```diff
@@ -2,6 +2,151 @@
...
+* Update copyright year in license.txt (#569)
+* Remove EOF file markers (#579)
+* Remove unnecessary unittest.main calls in tests (#581)
+* Replace on_trait_change with observe (#585, #590, #591, #625, #630)
+* Remove dummy return statements (#592, #593, #594)
+* Run black on the codebase (#599)
+* Remove outdated coverage badge (#623)
+* Use requires_redraw when possible (#626)
+* Deal with __getstate__ methods (#633)
+* Dont use traitsui.api as tui (#634)
```
Comment (rahulporuri) [inline CHANGES.txt:82]:
```suggestion
* Dont alias traitsui.api as tui (#634)
```

File: CHANGES.txt [hunk 17]
```diff
@@ -2,6 +2,151 @@
...
+* Import color palette from chaco.api directly (#675)
+* Remove backwards compatibily imports from enable (#679)
+* Move tests in "tests_with_backend" submodule into "tests" submodule (#683)
+* Rename test so it is discovered by unittest discover (#693)
+* Cleanup License files (#696)
+* Update "super" usage (#703, #706, #725)
+* Use a plot default method instead of overriding the init method in examples (#704, #709)
+* Replace custom "assert_raises" with "unittest.TestCase.assertRaises" (#705)
+* Concatenate change logs (#710)
+* re-export PolarLineRenderer and PolarMapper from chaco.api (#732)
```
Comment (rahulporuri) [inline CHANGES.txt:98]:
```suggestion
* Re-export PolarLineRenderer and PolarMapper from chaco.api (#732)
```

File: CHANGES.txt [hunk 18]
```diff
@@ -2,6 +2,151 @@
...
+* add editable option to ci install command (#733)
```
Comment (rahulporuri) [inline CHANGES.txt:99]:
```suggestion
* Add editable option to ci install command (#733)
```

File: CHANGES.txt [hunk 19]
```diff
@@ -2,6 +2,151 @@
...
+* remove calls to Component._draw_component (#737)
```
Comment (rahulporuri) [inline CHANGES.txt:100]:
```suggestion
* Remove calls to Component._draw_component (#737)
```

File: CHANGES.txt [hunk 20]
```diff
@@ -2,6 +2,151 @@
...
+* Add new chaco.plots submodule (#738)
+* use mock through unittest (#740)
```
Comment (rahulporuri) [inline CHANGES.txt:102]:
```suggestion
* Use mock through unittest (#740)
```

File: CHANGES.txt [hunk 21]
```diff
@@ -2,6 +2,151 @@
...
+* Consolidate overlays into "chaco.overlays" (#741)
+* Replace bg_color with bgcolor (#742)
+* Revive ScalyPlot (#744)
+
+Removals/Deprecations
```
Comment (rahulporuri) [inline CHANGES.txt:107]:
```suggestion
Removals/Deprecations

```

File: CHANGES.txt [hunk 22]
```diff
@@ -2,6 +2,151 @@
...
+* Remove uses of PyArray_FromDims (#503)
+* Remove unused _speedups module (#504)
+* remove SelectableOverlayPlotContainer and references to it (#574)
+* remove chaco.datamapper and subdivision code (#577)
```
Comment (rahulporuri) [inline CHANGES.txt:111]:
```suggestion
* Remove SelectableOverlayPlotContainer and references to it (#574)
* Remove chaco.datamapper and subdivision code (#577)
```

File: CHANGES.txt [hunk 23]
```diff
@@ -2,6 +2,151 @@
...
+* Delete codecov yaml configuration file (#622)
+* Remove examples for the vtk backend in chaco (#645)
+* Kill chaco.shell and related code (#667)
+* Remove objects/modules which explicitly raised a deprecation warning (#670)
+* Remove "chaco.scatter_markers" module (#677)
+* Delete "chaco.chaco_version" (#678)
+* Remove deprecated "ColorMapper._map" method (#684)
+* Delete sample_template.plt (#685)
+* Remove ChacoPlotItem / ChacoPlotEditor (#688)
+* delete scalar_image_function_inspector_old.py (#689)
```
Comment (rahulporuri) [inline CHANGES.txt:122]:
```suggestion
* Delete scalar_image_function_inspector_old.py (#689)
```

File: CHANGES.txt [hunk 24]
```diff
@@ -2,6 +2,151 @@
...
+* Officially deprecated VariableSizeScatterPlot (#691)
+* Remove PlotFrames (#692)
+* Delete "test_style.pss" (#697)
+* remove plot_template.py (#702)
+* remove a couple redundant old files (#711)
+* delete PopupablePlot (#717)
```
Comment (rahulporuri) [inline CHANGES.txt:128]:
```suggestion
* Remove plot_template.py (#702)
* Remove a couple redundant old files (#711)
* Delete PopupablePlot (#717)
```

File: CHANGES.txt (hunk 21 above)
Comment (rahulporuri) [inline CHANGES.txt:107]: I think this section needs to be moved to right below "Features/Enhancements/Improvements" - very very important given the large volume of code killed

File: CHANGES.txt [hunk 25]
```diff
@@ -2,6 +2,151 @@
...
+
+Build/CI
+
+* Update edm and install/config scripts (#492)
+* Use Cython to build extension modules (#502)
+* Remove Python 2.7, 3.5 testing, add PyQt 5 (#524)
+* Add source click option used on cron jobs to install ets packages from source (#532)
+* TST: Switch on default warning flag for CI test command (#537)
```
Comment (rahulporuri) [inline CHANGES.txt:136]:
```suggestion
* Switch on default warning flag for CI test command (#537)
```

File: CHANGES.txt [hunk 26]
```diff
@@ -2,6 +2,151 @@
...
+* Use PySide2 from EDM instead of PyPI and Use the latest EDM version (3.0.1) (#538)
+* Disable macOS builds on Travis CI (#543)
+* Fix cron job not installing additional dependencies (#544)
+* Add ci command for building documentation (#566)
+* run flake8 on CI but ignore all existing errors (#600)
```
Comment (rahulporuri) [inline CHANGES.txt:141]:
```suggestion
* Run flake8 on CI but ignore all existing errors (#600)
```

File: CHANGES.txt [hunk 27]
```diff
@@ -2,6 +2,151 @@
...
+* Add GitHub Actions CI workflow (#631)
+* Import version from installed package if the file doesnt exist (#644)
+* skip test_dont_crash_on_click on windows/pyqt (#666)
```
Comment (rahulporuri) [inline CHANGES.txt:144]:
```suggestion
* Skip test_dont_crash_on_click on windows/pyqt (#666)
```

**Review by rahulporuri — APPROVED** [review]
Mostly LGTM with one major comment - we need to move the Removals section almost to the top, to highlight it

**Pull request merged**